gramps/cli/grampscli.py

```diff
@@ -128,7 +128,7 @@ def _end_progress(self):
         """
         pass
 
-    def read_file(self, filename, username, password):
+    def read_file(self, filename, username, password, mode=None):
```

Comment:
Possibly default to "w" rather than None?

Reply:
Of course, you're right: we default to "w" unless the directory is read-only or it is set manually. Good point.

Reply:
Indeed the code is more concise now. Used `DMODE_W` and `DBMODE_R` while I was at it.